Rules/AvoidUsingGetWMIObject.cs

```diff
@@ -0,0 +1,127 @@
+﻿//
+// Copyright (c) Microsoft Corporation.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+using System;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using Microsoft.Windows.Powershell.ScriptAnalyzer.Generic;
+using System.ComponentModel.Composition;
+using System.Resources;
+using System.Globalization;
+using System.Threading;
+using System.Reflection;
+
+namespace Microsoft.Windows.Powershell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// AvoidUsingGetWMIObject: Verify that Get-WMIObject is not used
+    /// </summary>
+    [Export(typeof(IScriptRule))]
+    public class AvoidUsingGetWMIObject : IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Verify that Get-WMIObject is not used
+        /// </summary>
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            // Rule is applicable only when PowerShell Version is < 3.0, since Get-CIMInstance was introduced in 3.0
+            int majorPSVersion = GetPSMajorVersion(ast);
+            if (!(3 > majorPSVersion && 0 < majorPSVersion))
+            {
+                // Finds all CommandAsts.
+                IEnumerable<Ast> commandAsts = ast.FindAll(testAst => testAst is CommandAst, true);
+
+                // Iterate all CommandAsts and check the command name
+                foreach (CommandAst cmdAst in commandAsts)
+                {
+                    if (cmdAst.GetCommandName() != null && String.Equals(cmdAst.GetCommandName(), "get-wmiobject", StringComparison.OrdinalIgnoreCase))
```

Comment:
Remove-WMIObject help does not say that this cmdlet has been deprecated.  I will followup to see if we need to remove this also.